coalib/output/ConfWriter.py

```diff
@@ -89,15 +92,35 @@ def __write_key_val(self, keys, val):
 
         # Add escape characters as appropriate
         keys = [escape(key, chain(['\\'],
-                                  self.__key_value_delimiters,
+                                  self.__key_value_define_delimiters,
                                   self.__comment_separators,
                                   self.__key_delimiters,
                                   self.__section_override_delimiters))
                 for key in keys]
         val = escape(val, chain(['\\'], self.__comment_separators))
 
-        self.__file.write((self.__key_delimiter + ' ').join(keys) + ' ' +
-                          self.__key_value_delimiter + ' ' + val + '\n')
+        append_keys = []
+        for key in keys:
+            if (section.defaults and
+                    (key in section.defaults and
+                     (val.startswith(str(section.defaults[key])+',')))):
```

Comment:
so if I have this:

```ini
[defaults]
a = 4
b = 4,5
[defaults.new]
a = 4,5,6
b = 4,5,6
```

I think your algorithm would transform it to:

```ini
[defaults]
a = 4
b = 4,5
[defaults.new]
a, b += 5,6
```

maybe we should have like a generator expression to get the first append string and then check for which keys it's valid, that should be a good enough heuristic